src/WebJobs.Script/Rpc/FunctionRegistration/FunctionDispatcherLoadBalancer.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+
+namespace Microsoft.Azure.WebJobs.Script.Rpc
+{
+    internal class FunctionDispatcherLoadBalancer : IFunctionDispatcherLoadBalancer
+    {
+        private int _maxProcessCount = 1;
+        private static int _counter = 1;
+
+        internal FunctionDispatcherLoadBalancer(int maxProcessCount)
+        {
+            _maxProcessCount = maxProcessCount > _maxProcessCount ? maxProcessCount : _maxProcessCount;
+        }
+
+        public ILanguageWorkerChannel GetLanguageWorkerChannel(IEnumerable<ILanguageWorkerChannel> languageWorkerChannels)
+        {
+            if (languageWorkerChannels == null)
+            {
+                throw new ArgumentNullException(nameof(languageWorkerChannels));
+            }
+
+            var currentNumberOfWorkers = languageWorkerChannels.Count();
+
+            if (currentNumberOfWorkers == 0)
+            {
+                throw new InvalidOperationException($"Number of language workers:{currentNumberOfWorkers}");
+            }
+            if (currentNumberOfWorkers > _maxProcessCount)
+            {
+                throw new InvalidOperationException($"Number of language workers exceeded:{currentNumberOfWorkers} exceeded maxProcessCount: {_maxProcessCount}");
+            }
+            ILanguageWorkerChannel workerChannel = languageWorkerChannels.ElementAt(_counter % currentNumberOfWorkers);
+            _counter = _counter < currentNumberOfWorkers ? Interlocked.Increment(ref _counter) : 1;
```

Comment:
I'm a little suspicious of this, since it seems like it could lead to some channels getting disproportionate workloads (since all threads are referencing the same static _counter value). I'm imagining a scenario where two requests are working concurrently and counter = 1 with currentNumberOfWorkers = 2. so worker channel at index 1 gets chosen for both. then, both choose to increment because _counter (1) < currentNumberOfWorkers (2). _counter increases to 2 then 3, atomically. then next execution, the language worker at element 1 is chosen again

I'm not totally sure, but I wonder if logic like this might help avoid overloading one or a few workers?

```
// checks for invalid operation exceptions

var counter = Interlocked.Increment(ref _counter);
// something to make sure we don't go over the max int value?
if (_counter > int.MaxValue - 1000) {
    _counter = 1;
}
return languageWorkerChannels.ElementAt(_counter % currentNumberOfWorkers);
```

Reply:
As language workers gets initialized, invocations start to evenly distribute. For the example above, yes when the system is initializing there might be a point of time where workerChannel (1) will get more invocations, but once all the language workers are fully initialized, distribution would be even.

Reply:
If this gets called for every invocation, you could have a lot of simultaneous evaluations of  `_counter < currentNumberOfWorkers`. That could lead to some of the things @mhoeger mentions above. It seems like worker 1 will end up getting more traffic than others. 

@mhoeger's solution is good, but we don't even need to worry about MaxValue. Interlocked.Incremement will wrap from Int32.MaxValue -> Int32.MinValue. So we can just check if workerIndex is negative, just take the -workerIndex. That should balance things better.

Reply:
Makes sense. Thank you both!